rpcs3/Emu/Cell/SPUThread.cpp

```diff
@@ -1500,55 +1500,101 @@ bool spu_thread::do_dma_check(const spu_mfc_cmd& args)
 
 bool spu_thread::do_list_transfer(spu_mfc_cmd& args)
 {
-	struct list_element
+	// Amount of elements to fetch in one go
+	constexpr u32 fetch_size = 6;
+
+	struct alignas(8) list_element
 	{
 		be_t<u16> sb; // Stall-and-Notify bit (0x8000)
 		be_t<u16> ts; // List Transfer Size
 		be_t<u32> ea; // External Address Low
-	} item{};
+	};
 
-	while (args.size)
+	union
 	{
-		if (UNLIKELY(item.sb & 0x8000))
-		{
-			ch_stall_mask |= utils::rol32(1, args.tag);
+		list_element items[fetch_size];
+		alignas(v128) char bufitems[sizeof(items)];
+	};
 
-			if (!ch_stall_stat.get_count())
+	spu_mfc_cmd transfer;
+	transfer.eah  = 0;
+	transfer.tag  = args.tag;
+	transfer.cmd  = MFC(args.cmd & ~MFC_LIST_MASK);
+
+	args.lsa &= 0x3fff0;
+
+	u32 index = fetch_size;
+
+	// Assume called with size greater than 0
+	while (true)
+	{
+		// Check if fetching is needed
+		if (index == fetch_size)
+		{
+			const auto src = _ptr<void>(args.eal & 0x3fff8);
+			if ((uptr)src % alignof(v128))
```

Comment:
Can union be replaced with simple `alignas(16) list_element items[fetch_size];`, and the whole alignment check with two patches with single memcpy?